contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,199 @@
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function _tokenDecimalsMultiplier(address tokenAddress) public pure returns (uint256) {
+        uint256 decimalsNeeded = 18 - IJoeERC20(tokenAddress).decimals();
+        return uint256(1 * (10**decimalsNeeded));
+    }
+
+    function getReserveUSD(IJoePair pair) public view returns (uint256) {
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves(); // reserve0, reserve1 are 18 decimals
+
+        uint256 token0PriceInAvax = getPriceInAvax(token0Address); // 18
+        uint256 token1PriceInAvax = getPriceInAvax(token1Address); // 18
+        uint256 token0ReserveUSD = (reserve0.mul(_tokenDecimalsMultiplier(token0Address))).mul(token0PriceInAvax).mul(
+            getAvaxPrice()
+        ); // 18.mul(18).mul(18) = 54 decimals
+        uint256 token1ReserveUSD = (reserve1.mul(_tokenDecimalsMultiplier(token1Address))).mul(token1PriceInAvax).mul(
+            getAvaxPrice()
+        ); // 54
+
+        uint256 reserveUSD = (token0ReserveUSD.add(token1ReserveUSD)) / uint256(1e36);
+
+        return reserveUSD; // 18
+    }
+
+    struct FarmPair {
+        address lpAddress;
+        address token0Address;
+        address token1Address;
+        string token0Symbol;
+        string token1Symbol;
+        uint256 reserveUSD;
+        uint256 totalSupply;
+        address chefAddress;
+        uint256 chefBalance;
+        uint256 chefTotalAlloc;
+        uint256 chefJoePerSec;
+    }
+
+    function getFarmPairs(address[] calldata pairAddresses, address chefAddress)
+        public
+        view
+        returns (FarmPair[] memory)
+    {
+        FarmPair[] memory farmPairs = new FarmPair[](pairAddresses.length);
+
+        for (uint256 i = 0; i < pairAddresses.length; i++) {
+            // get pair information
+            IJoePair lpToken = IJoePair(pairAddresses[i]);
+            address lpAddress = address(lpToken);
+            address token0Address = lpToken.token0();
+            address token1Address = lpToken.token1();
+            farmPairs[i].token0Address = token0Address;
+            farmPairs[i].token1Address = token1Address;
+            farmPairs[i].token0Symbol = IJoeERC20(token0Address).symbol();
+            farmPairs[i].token1Symbol = IJoeERC20(token1Address).symbol();
+
+            // calculate reserveUSD of lp
+            farmPairs[i].reserveUSD = getReserveUSD(lpToken); // 18
+
+            // calculate total supply of lp
+            farmPairs[i].totalSupply = lpToken.totalSupply().mul(_tokenDecimalsMultiplier(lpAddress));
+
+            // get masterChef data
+            uint256 balance = lpToken.balanceOf(chefAddress);
+            farmPairs[i].chefBalance = balance.mul(_tokenDecimalsMultiplier(lpAddress));
```

Comment:
Same here, should we be scaling by 1e18? If so, then rename to `chefBalanceScaled`

Reply:
Ahh I see, I was scaling everything to be 18 decimals because I wasn't sure how else we would know the decimals being returned here. Would we want to return the decimals for the pair here as well?

Reply:
Really up to you and how it's being used on the frontend. Either way works, but needs to be crystal clear

Reply:
Gotcha 👍 I'll stick with scaling it and rename this the variables appropriately

Reply:
Gotcha 👍 I'll stick with scaling it and rename the variables appropriately